src/main/java/fi/jubic/quanta/external/importer/csv/CsvImporter.java

```diff
@@ -165,6 +179,9 @@ public DataSample getSample(DataSeries dataSeries, int rows) {
                     .setData(sampleRows)
                     .build();
         }
+        catch (InputException e) {
```

Comment:
`InputException` is a subclass of `ApplicationException`. There's no need to catch it and rethrow. Just make sure the message string is included on the first go.

Reply:
Change to `() -> new InputException("Can not import empty CSV file.")`